internal/resources/metal/vlan/resource_test.go

```diff
@@ -63,6 +77,89 @@ func TestAccMetalVlan_metro(t *testing.T) {
 						"equinix_metal_vlan.foovlan", "metro", metro),
 				),
 			},
+			{
+				Config:   testAccCheckMetalVlanConfig_metro(rs, strings.ToUpper(metro), "tfacc-vlan"),
+				PlanOnly: true,
+			},
+			{
+				Config: testAccCheckMetalVlanConfig_metro(rs, upperMetro, "tfacc-vlan"),
+				ConfigPlanChecks: resource.ConfigPlanChecks{
+					PreApply: []plancheck.PlanCheck{
+						plancheck.ExpectResourceAction("equinix_metal_vlan.foovlan", plancheck.ResourceActionDestroyBeforeCreate),
+					},
+				},
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckMetalVlanExists("equinix_metal_vlan.foovlan", &vlan),
+					resource.TestCheckResourceAttr(
+						"equinix_metal_vlan.foovlan", "description", "tfacc-vlan"),
+					resource.TestCheckResourceAttr(
+						"equinix_metal_vlan.foovlan", "metro", upperMetro),
+				),
+			},
+			{
+				Config:   testAccCheckMetalVlanConfig_metro(rs, strings.ToLower(upperMetro), "tfacc-vlan"),
+				PlanOnly: true,
```

Comment:
This test step confirms that there are no changes to apply after changing the metro from `DA` to `da`

Reply:
And here you could use metro, right?

Reply:
I can see how this test is hard to reason about; in the latest commit I tried changing the names of the `upperMetro` and `metro` variables in an attempt to clarify intent/usage.  Let me know if that helped or no.